### PR TITLE
Don't release action if it's not pressed

### LINKED ISSUE #30722: The last Input.is_action_just_released() fires a 2nd time when app focus lost
the last Input.is_action_just_released() fires a 2nd time when app focuses lost.
Ie  say the last call was checking if F10 was pressed :

	if Input.is_action_just_released("input_key_F10"): #  /set to F10  
		print("is_action_just_released."   )

you will see the print 1 time to console in the editor, or in the exported apps debug.
now alt-tab, or simply click on another app, any other app and you will see it fire again and a 2nd line will appear as the print() is reached a second time.

for my use, I was able to use Input.is_action_just_pressed(). It works as expected in the case that helps anyone else who needs a quick fix.

**Godot version:**
Nightly 3.2, Dev as of today 

**OS/device including version:**
W10, Ryzen 3 1200, 16GB ram, Zotac GTX1050ti, standard keyboard layout 

**Issue description:**
the last Input.is_action_just_released() is true a 2nd time on app loss of focus, the moment you click in/on another app.

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #30722

### CODE REVIEW

**Comment by Xrayez** [comment]
Would this interfere somehow with the fact that the gameplay can be simulated via parsing input actions via code, even if the window focus is lost? Though I haven't been able to pass these kind of virtual input events when the window is out of focus, so this might qualify as a separate issue. Just a thought.

**Comment by KoBeWi** [comment]
> Would this interfere somehow with the fact that the gameplay can be simulated via parsing input actions via code, even if the window focus is lost?

It has nothing to do with this change. You still can (or can't) do this, the PR just prevents from releasing something twice when it shouldn't be.

(now that I think of it, the opposite action might still be possible, i.e. pressing something twice without releasing it...)

**Comment by Xrayez** [comment]
> (now that I think of it, the opposite action might still be possible, i.e. pressing something twice without releasing it...)

Oh well I thought that it's possible to press action twice without releasing it since you can assign multiple keys to the same action. The opposite was also on my mind because when you press the same action twice using multiple keys, you'd inevitably get a sequence of released actions, but seems like it's not the case as I expected it to be.

**Comment by Xrayez** [comment]
Upon further investigation I've figured that it's indeed possible to press an action twice and then release it twice with this code snippet:

```gdscript
var press_count = 0


func _input(event):
	if event.is_action_pressed('ui_accept'):
		press_count += 1
		print(press_count)
	elif event.is_action_released('ui_accept'):
		press_count -= 1
		print(press_count)
```

After <kbd>Enter</kbd> > <kbd>Space</kbd> pressed and then <kbd>Enter</kbd> > <kbd>Space</kbd> released I get the following output:
```
1
2
1
0
```

I wasn't able to do so previously because I used `just_pressed/released` variants of the methods, but seems like it's doable and shouldn't be restricted (?).

**Comment by akien-mga** [comment]
Yeah I think the check should go in `release_pressed_events()` to make it do what it says on the tin (only release "pressed" events). `action_release()` itself doesn't need to be restricted.

**Comment by KoBeWi** [comment]
Ok, changed. That's what I wanted to do in the beginning actually, but wasn't sure what and where to put a condition.

File: main/input_default.cpp [hunk 1]
```diff
@@ -686,7 +686,8 @@ void InputDefault::release_pressed_events() {
 	_joy_axis.clear();
 
 	for (Map<StringName, InputDefault::Action>::Element *E = action_state.front(); E; E = E->next()) {
-		action_release(E->key());
+		if (action_state[E->key()].pressed)
```
Comment (akien-mga) [inline main/input_default.cpp]: Since `E` is already an element from `action_state`, wouldn't `E->get().pressed` be the more direct way to access this?

Comment (KoBeWi) [reply main/input_default.cpp]: Changed.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**